src/services/identity/SitesService.ts

```diff
@@ -194,6 +285,41 @@ class SitesService {
       where: { id: updateParams.id },
     })
   }
+
+  async getSiteInfo(sessionData: UserWithSiteSessionData): Promise<SiteInfo> {
+    const {
+      staging: stagingUrl,
+      prod: prodUrl,
+    } = await this.getUrlsOfSite(sessionData, ["staging", "prod"])
+
+    const stagingCommit = ((await this.gitHubService.getLatestCommitOfBranch(
+      sessionData,
+      "staging"
+    )) as unknown) as GitHubCommitData
```

Comment:
this is a dangerous cast. we should validate the properties and use an assertion.

Reply:
Added an assertion function in 9ca0838.